magda-io/magda
# Issue/2421 Make it possible to run in single-tenant mode without a tenant api

### What this PR does

Fixes #2421 Make it possible to run in single-tenant mode without a tenant api
- add helm chart global switch `enableMultiTenants` to turn multi-tenant mode on/off
  - Gateway used to have a local `enableMultiTenants` switch. It's now replaced with this global one as there shouldn't be a use case that sub modules run in different MultiTenants mode.
- disabled tenant-db & tenant-api when enableMultiTenants == false
- remove unused `tenantId` arg from minion
  - Although a `tenantId` parameter had been passed to minions, all minions used a fixed `-1` value
- Fixed test cases & added `enableMultiTenants` to property test
- Made gateway skip tenant-api router & status prob

### Testing

- [x] Deploy with  `enableMultiTenants` = false to a clean env
- [x] Deploy version `0.0.55-RC1` (pre- multi-tenant version) and then upgrade to latest version with `enableMultiTenants` = false
  - Remove `image.repository` key & `image.tag` key in `minikube-dev.yml`
  - `helm upgrade --install --timeout 9999 --wait --namespace magda --version 0.0.55-RC1 -f deploy/helm/minikube-dev.yml magda magda-io/magda`
  - Test connector, minions & searching
  - Set `image.repository`=`localhost:5000/data61` & image.tag`=`latest`
  - `helm upgrade --install --timeout 9999 --wait --namespace magda -f deploy/helm/minikube-dev.yml magda deploy/helm/magda`
  - Make sure tenant-api not created
  - Test conenctor, moinions & searching
- [x] Deploy with  `enableMultiTenants` = true to a clean env

### Checklist
